### PR TITLE
refactor(tooltip): 增加 adjustPosition 回调参数

### PR DESCRIPTION
**Pull request by xingwanying**
### 👀 PR includes

🎨 Enhance

- [ ] Code style optimization
- [x] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

### 📝 Description

* 将鼠标点击事件透出给 adjustPosition 参数
* 更新对应文档

### 🔍 Self Check before Merge

- [x] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [x] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/common/interface/tooltip.ts [hunk 1]
```diff
@@ -154,13 +154,20 @@ export interface BaseTooltipConfig<T = TooltipContentType> {
...
+  ) => TooltipPosition;
   // Custom tooltip content
   readonly renderTooltip?: (spreadsheet: SpreadSheet) => BaseTooltip;
   // Custom tooltip mount container
   readonly getContainer?: () => HTMLElement;
 }
 
+export interface TooltipPositionInfo {
+  position: TooltipPosition;
+  event: CanvasEvent | MouseEvent;
```
Comment (xingwanying) [inline packages/s2-core/src/common/interface/tooltip.ts:168]: 只透出 event 即可，能拿到当前点击的 cell 信息，且有可能点击到的是 icon

File: packages/s2-core/src/sheet-type/spread-sheet.ts [hunk 2]
```diff
@@ -260,6 +260,7 @@ export abstract class SpreadSheet extends EE {
 
     this.tooltip.show?.({
       ...showOptions,
+      event,
```
Comment (lijinke666) [inline packages/s2-core/src/sheet-type/spread-sheet.ts:263]: showOptions 已经有 event 了, 冗余了

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**